app/services/invoice_renderer.rb

```diff
@@ -0,0 +1,29 @@
+class InvoiceRenderer
+  def render_to_string(order)
+    renderer.render_to_string(args(order))
```

Comment:
This class would clearly benefit from moving `order` to an ivar populated in the constructor IMO

Reply:
I don't see it as clearly. I see great benefit in declaring all input variables for each method. It makes refactoring easier. It also allow to call the same renderer with different orders. The BulkInvoiceService should probably do that.